Utilities/XrdAdaptor/src/XrdStatistics.cc

```diff
@@ -0,0 +1,179 @@
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/MessageLogger/interface/JobReport.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+
+#include "XrdRequest.h"
+#include "XrdStatistics.h"
+
+#include <chrono>
+
+using namespace XrdAdaptor;
+
+
+std::atomic<XrdSiteStatisticsInformation*> XrdSiteStatisticsInformation::m_instance;
+
+
+XrdStatisticsService::XrdStatisticsService(const edm::ParameterSet &iPS, edm::ActivityRegistry &iRegistry)
+{
+    if (iPS.getUntrackedParameter<bool>("reportToFJR", false))
+    {
+        XrdSiteStatisticsInformation::createInstance();
+    }
+
+    iRegistry.watchPostEndJob(this, &XrdStatisticsService::postEndJob);
+}
+
+
+void XrdStatisticsService::postEndJob()
+{
+    edm::Service<edm::JobReport> reportSvc;
+    if (!reportSvc.isAvailable()) {return;}
+
+    XrdSiteStatisticsInformation *instance = XrdSiteStatisticsInformation::getInstance();
+    if (!instance) {return;}
+
+    for (std::shared_ptr<XrdSiteStatistics> const &stats : instance->m_sites)
+    {
+        stats->recomputeProperties();
```

Comment:
```recomputeProperties``` is the only thread unsafe function in this class. Given you compute and then use the values immediately, I see no reason for them to be stored in a member data. Instead, pass in your ```std::map``` as an argument to ```computeProperties``` and then pass that map to ```reportPerformanceForModule```. It will be just as efficient but make the ```XrdSiteStatistics``` fully thread safe.